### PR TITLE
Contents.plist: use right exception when missing

### PR DESCRIPTION
**Pull request by madig**
Need to figure out the tests. `test_validators_plist_ufo2_contents_missing_file_fail` and `test_validators_plist_ufo3_contents_missing_file_fail` don't actually test for failure?

### CODE REVIEW

**Comment by madig** [comment]
lgtm is somehow out of date with its dependencies?

File: lib/ufolint/validators/plistvalidators.py
Comment (chrissimpkins) [inline lib/ufolint/validators/plistvalidators.py:295]: For (my) future reference, this is the new parameter that was added in fonttools v4.18.0

File: lib/ufolint/validators/plistvalidators.py
Comment (chrissimpkins) [inline lib/ufolint/validators/plistvalidators.py:309]: How do you suggest that we refactor the sacrilege here?

**Review by chrissimpkins — APPROVED** [review]

**Comment by chrissimpkins** [comment]
OK to ignore it.  I'm not sure why it flags it but I can see that it was added in fonttools 4.18.0 and your unit tests are passing.  This gtg?

File: lib/ufolint/validators/plistvalidators.py
Comment (madig) [reply lib/ufolint/validators/plistvalidators.py:309]: Oh, no, I consider unregistered glif files a sacrilege that merits an error even when it doesn't prevent compilation.

**Comment by madig** [comment]
Yes! If you sign off my changes to the existing tests? I found it odd that they tested for seccuess when they should be expecting failure?

**Comment by chrissimpkins** [comment]
> Yes! If you sign off my changes to the existing tests? I found it odd that they tested for success when they should be expecting failure?

Running tests locally to have a look.  I am seeing this message:

> tests/test_runner.py: 17 warnings
  /Users/chris/code/ufolint/.tox/py38/lib/python3.8/site-packages/ufolint/controllers/runner.py:356: DeprecationWarning: The 'formatVersion' attribute is deprecated; use the 'formatVersionTuple'
    self.ufoversion = ufolib_reader.formatVersion

This going away soon?

**Comment by chrissimpkins** [comment]
> If you sign off my changes to the existing tests? 

LGTM.  I think that this is a case of a missing test on a test :)  Since it was not implemented in fonttools and didn't raise an exception, I didn't recognize that my copy/paste code was not testing against the fail case.  Bad source passed because there was previously no exception.  It failed in the commit that included updated fonttools and ufolint source before the test changes were added: https://github.com/source-foundry/ufolint/runs/1517236056

**Comment by chrissimpkins** [comment]
Posted the deprecation issue in https://github.com/source-foundry/ufolint/issues/88.  We are ready to merge here.  I will update the deprecated source.  Thanks Niko!!

**Comment by madig** [comment]
> This going away soon?

Mh, no idea, but should be easy to fix.

**Comment by madig** [comment]
PS: can't merge so I'll leave that to you :D

**Comment by chrissimpkins** [comment]
> PS: can't merge so I'll leave that to you :D

Sorry, I think that the permission system changed since I last added people to this org.  Or maybe it went away when I got rid of the tool team that auto added you and Cosimo as watch on any new repository.  I didn't want force you to get pinged with all of the notifications.

I added you to a new ["Maintainers" team](https://github.com/orgs/source-foundry/teams/maintainers/members) and added this repo to it.  I'll add the rest of the tool repos so that you have maintain level access to them if you need it.

**Comment by chrissimpkins** [comment]
https://pypi.org/project/ufolint/1.2.0/

Thanks Niko!!

**Comment by chrissimpkins** [comment]
@felipesanches @graphicore we had to make the fonttools dependency a hard fonttools >= 4.18.0+ in ufolint to introduce a new UFO source validation committed in this PR.  This was released in v1.2.0.  Is this a problem for fontbakery?

**Comment by graphicore** [comment]
I don't think there's a general problem, at least not in the way and context we use font bakery usually (virtual environments etc.). Similar things have been done before in Font Bakery, I see [in setup.py](https://github.com/googlefonts/fontbakery/blob/master/setup.py):

>         'fontTools[ufo,lxml,unicode]>=3.34',  # 3.34 fixed some CFF2 issues, including calcBounds

From the README of FontTools:

> FontTools 4.x requires Python 3.6 or later. FontTools 3.x requires Python 2.7 or later.

Font Bakery is Python 3.6+ as well.

But, @felipesanches is the better person to answer this.

**Pull request merged**
